Apps/FCM/ios/FCMSampleApp/AppDelegate.swift

```diff
@@ -0,0 +1,131 @@
+import UIKit
+import React
+import CioMessagingPushFCM
+import CioTracking
+import Firebase
+import FirebaseMessaging
+
+@UIApplicationMain
+class AppDelegate: UIResponder, UIApplicationDelegate {
+  
+  let moduleName: String = "FCMSampleApp"
+  let initialProps: [AnyHashable: Any] = [:]
+  var window: UIWindow?
+
+  func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+    // Add custom initialization code here
+    // This fixes an issue in which the React Native SDK identifies a profile, but an iOS device isn't added to the profile.
+    // Important to call this code before calling `registerForRemoteNotifications`.
+    CustomerIO.initialize(siteId: Env.siteId, apiKey: Env.apiKey, region: .US) { config in
+      config.autoTrackPushEvents = true
+      config.logLevel = .debug
+    }
+
+    // Configure Firebase
+    FirebaseApp.configure()
+    
+    var modifiedLaunchOptions = launchOptions
+
+    if let options = launchOptions,
+       options[UIApplication.LaunchOptionsKey.url] == nil,
+       let pushContent = options[UIApplication.LaunchOptionsKey.remoteNotification] as? [String: Any],
+       let cioContent = pushContent["CIO"] as? [String: Any],
+       let pushContent = cioContent["push"] as? [String: Any],
+       let initialLink = pushContent["link"] as? String {
+      modifiedLaunchOptions![UIApplication.LaunchOptionsKey.url] = URL(string: initialLink)
+    }
+    
+    let bridge = RCTBridge(delegate: self, launchOptions: modifiedLaunchOptions)!
+    let rootView = RCTRootView(bridge: bridge, moduleName: moduleName, initialProperties: initialProps)
+    rootView.backgroundColor = UIColor(red: 1.0, green: 1.0, blue: 1.0, alpha: 1)
+    self.window = UIWindow(frame: UIScreen.main.bounds)
+    let rootViewController = UIViewController()
+    rootViewController.view = rootView
+    self.window?.rootViewController = rootViewController
+    self.window?.makeKeyAndVisible()
+
+    // Now that the Firebase and Customer.io SDK's are initialized, follow the rest of the required steps for the FCM push setup.
+    UNUserNotificationCenter.current().delegate = self
+    
+    // Set FCM messaging delegate
+    Messaging.messaging().delegate = self
+    
+    // Uncomment only if you want to register for remote push when the app starts
+    // UIApplication.shared.registerForRemoteNotifications()
```

Comment:
Why is this commented out ? I might have missed it but are we registering remote notifications somewhere else ?

Reply:
We already do this in our [react native SDK](https://github.com/customerio/customerio-reactnative/blob/153228cbc405d28a3c0134f405478845afc21099/ios/CustomerioReactnative.swift#L60-L62). Do we still need this? I think we should remove this completely from here as registering from the SDK should probably do the job.

Reply:
Because this exact code is in the RN SDK, I don't see a reason to keep this code in this sample app at all. It would be a good idea to double check our RN docs to remove this recommendation if we still do that.

Reply:
Agreed. It is good to remove this snippet from here and docs as well. The comment on the top is confusing too.

Reply:
Removed this completely in Objective-C version.